crypto/bn256/bn256_fuzz.go

```diff
@@ -54,73 +64,50 @@ func FuzzAdd(data []byte) int {
 	if !bytes.Equal(rc.Marshal(), rg.Marshal()) {
 		panic("add mismatch")
 	}
-	return 0
+	return 1
 }
 
 // FuzzMul fuzzez bn256 scalar multiplication between the Google and Cloudflare
 // libraries.
 func FuzzMul(data []byte) int {
-	// Ensure we have enough data in the first place
-	if len(data) != 96 {
+	input := bytes.NewReader(data)
+	pc, pg := getG1Points(input)
+	if pc == nil {
 		return 0
 	}
-	// Ensure both libs can parse the curve point
-	pc := new(cloudflare.G1)
-	_, errc := pc.Unmarshal(data[:64])
-
-	pg := new(google.G1)
-	_, errg := pg.Unmarshal(data[:64])
-
-	if (errc == nil) != (errg == nil) {
-		panic("parse mismatch")
-	} else if errc != nil {
+	// Add the two points and ensure they result in the same output
+	remaining := input.Len()
+	if remaining == 0 {
 		return 0
 	}
-	// Add the two points and ensure they result in the same output
+	buf := make([]byte, remaining)
```

Comment:
The scalar is max 32 bytes but here we just take the remaining input. I think the remaining input is fine since it covers more ground than the precompiled contracts